web_ui/frontend/app/registry/page.tsx

```diff
@@ -18,29 +18,112 @@
 
 "use client"
 
-import RateGraph from "@/components/graphs/RateGraph";
-import StatusBox from "@/components/StatusBox";
+import {Box, Button, Grid, Typography, Skeleton, Alert, Collapse} from "@mui/material";
+import React, {useEffect, useMemo, useState} from "react";
 
-import {TimeDuration} from "@/components/graphs/prometheus";
-
-import {Box, Grid, Typography} from "@mui/material";
-import FederationOverview from "@/components/FederationOverview";
-import {ServerTable} from "@/components/ServerTable";
-import NamespaceTable from "@/components/NamespaceTable";
+import {PendingCard, Card, NamespaceCardSkeleton, CreateNamespaceCard} from "@/components/Namespace";
+import Link from "next/link";
+import {Namespace, Alert as AlertType} from "@/components/Main";
+import UnauthenticatedContent from "@/components/layout/UnauthenticatedContent";
+import {Authenticated, getAuthenticated, isLoggedIn} from "@/helpers/login";
 
 
 export default function Home() {
 
+    const [data, setData] = useState<Namespace[] | undefined>(undefined);
+    const [alert, setAlert] = useState<AlertType | undefined>(undefined)
+    const [authenticated, setAuthenticated] = useState<Authenticated | undefined>(undefined)
+
+    const getData = async () => {
+
+        let data : Namespace[] = []
+
+        const url = new URL("/api/v1.0/registry_ui/namespaces", window.location.origin)
+
+        const response = await fetch(url)
+        if (response.ok) {
+            const responseData: Namespace[] = await response.json()
+            responseData.sort((a, b) => a.id > b.id ? 1 : -1)
+            responseData.forEach((namespace) => {
+                if (namespace.prefix.startsWith("/cache")) {
+                    namespace.type = "cache"
+                } else {
+                    namespace.type = "origin"
+                }
+            })
+            data = responseData
+        }
+
+        return data
+    }
+
+    const _setData = async () => {setData(await getData())}
+
+    useEffect(() => {
+        _setData();
+        (async () => {
+            if(await isLoggedIn()){
+                setAuthenticated(getAuthenticated() as Authenticated)
+            }
+        })();
+    }, [])
+
+    const pendingData = useMemo(
+        () => data?.filter(
+            (namespace) => namespace.admin_metadata.status === "Pending" &&
+                (authenticated?.user == namespace.admin_metadata.user_id || authenticated?.role == "admin")
+                ), [data, authenticated]
+    )
+    const approvedCacheData = useMemo(
```

Comment:
For admin users, they should be able to see a list of denied registrations. For regular users, it should be also helpful to see _their_ denied registrations. Right now denied registrations are gone.

Reply:
Agreed but no time this round to implement that.